src/OpenTelemetry/Batch.cs

```diff
@@ -46,7 +48,23 @@ internal Batch(CircularBuffer<T> circularBuffer, int maxSize)
 
             this.item = null;
             this.circularBuffer = circularBuffer ?? throw new ArgumentNullException(nameof(circularBuffer));
-            this.maxSize = maxSize;
+            this.addedCount = circularBuffer.AddedCount;
+
+            long removedCount = circularBuffer.RemovedCount;
+            this.targetCount = removedCount + Math.Min(maxSize, this.addedCount - removedCount);
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            if (this.circularBuffer != null)
+            {
+                // Drain anything left in the batch.
+                while (this.circularBuffer.RemovedCount < this.addedCount)
```

Comment:
It only drains up to the batch size.

```suggestion
                while (this.circularBuffer.RemovedCount < this.targetCount)
```

Reply:
Good catch. I updated and added a test case to make sure it only drains up to maxSize.